cartridge/webui.lua

```diff
@@ -63,6 +89,29 @@ local function init(httpd)
         callback = module_name .. '.get_blacklist',
     })
 
+    -- LSP
+    local lsp_handler
+    if options.lsp_enabled == true then
+        lsp_handler = lsp.create_websocket_handler()
+    else
+        lsp_handler = function()
+            local err = LspError:new('LSP support is disabled')
+            return http_finalize_error(501, err)
+        end
+    end
+
+    httpd:route({
+        path = '/admin/lsp',
+        method = 'GET'
+    }, function(req)
+        if not auth.authorize_request(req) then
+            local err = errors.new('LspError', 'Unauthorized')
+            return http_finalize_error(401, err)
+        end
+
+        return lsp_handler(req)
```

Comment:
Your handler is an endless loop and you are never return the response to WebUI. It's wrong.